huaweicloud/services/dli/resource_huaweicloud_dli_template_sql.go

```diff
@@ -0,0 +1,283 @@
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceSQLTemplate() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceSQLTemplateCreate,
+		UpdateContext: resourceSQLTemplateUpdate,
+		ReadContext:   resourceSQLTemplateRead,
+		DeleteContext: resourceSQLTemplateDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The name of the SQL template.`,
+			},
+			"sql": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The statement of the SQL template.`,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `The description of the SQL template.`,
+			},
+			"group": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `The group of the SQL template.`,
+			},
+			"owner": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The user ID of owner.`,
+			},
+		},
+	}
+}
+
+func resourceSQLTemplateCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// createSQLTemplate: create a SQLTemplate.
+	var (
+		createSQLTemplateHttpUrl = "v1.0/{project_id}/sqls"
+		createSQLTemplateProduct = "dli"
+	)
+	createSQLTemplateClient, err := cfg.NewServiceClient(createSQLTemplateProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DLI Client: %s", err)
+	}
+
+	createSQLTemplatePath := createSQLTemplateClient.Endpoint + createSQLTemplateHttpUrl
+	createSQLTemplatePath = strings.ReplaceAll(createSQLTemplatePath, "{project_id}", createSQLTemplateClient.ProjectID)
+
+	createSQLTemplateOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	createSQLTemplateOpt.JSONBody = utils.RemoveNil(buildCreateSQLTemplateBodyParams(d, cfg))
+	createSQLTemplateResp, err := createSQLTemplateClient.Request("POST", createSQLTemplatePath, &createSQLTemplateOpt)
+	if err != nil {
+		return diag.Errorf("error creating SQLTemplate: %s", err)
+	}
+
+	createSQLTemplateRespBody, err := utils.FlattenResponse(createSQLTemplateResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("sql_id", createSQLTemplateRespBody)
+	if err != nil {
+		return diag.Errorf("error creating SQLTemplate: ID is not found in API response")
+	}
+	d.SetId(id.(string))
+
+	return resourceSQLTemplateRead(ctx, d, meta)
+}
+
+func buildCreateSQLTemplateBodyParams(d *schema.ResourceData, _ *config.Config) map[string]interface{} {
```

Comment:
this func is same to buildUpdateSQLTemplateBodyParams, please them into one func